src/test/scala/exercise3/KarplusSpec.scala

```diff
@@ -0,0 +1,43 @@
+package exercise3
+
+import zio._
+import zio.test._
+import zio.test.Assertion._
+import zio.test.environment.TestRandom
+import exercise3.KarplusStrong._
+
+object KarplusSpec extends DefaultRunnableSpec {
+
+  def spec = suite("KarplusSpec whiteNoise")(
+    testM("Queue shouldn't be empty by default") {
+      for {
+        queue <- whiteNoise()
+      } yield assert(queue.isEmpty)(equalTo(false))
+    },
+
+    testM("Queue-Element is instance of Some") {
+      for {
+        queue <- whiteNoise()
+      } yield  assert(queue.front()) (isSome)
+    },
+
+    testM("Queue value should be between -0.5 and 0.5") {
+      val randomDoubles = List(0.5, 0.2, 0.3)
+      val expectedRandoms = List(0.25, 0.1, 0.15)
+      for {
+        _  <- TestRandom.feedDoubles(randomDoubles(0), randomDoubles(1), randomDoubles(2))
+        queue <- whiteNoise(3, 0.5)
+      } yield assert(expectedRandoms)(equalTo(List(queue.front().get, queue.dequeue().get.front().get, queue.dequeue().get.dequeue().get.front().get)))
```

Comment:
If your Stack and Queue were case classes you could just assert the queue is equal to

```Queue.enqueue(0.25).enqueue(0.1).enqueue(0.15)```
which is a bit less abominable, but since there is no companion object for the Queue, the better way.

If we had something like 
```scala
object Queue{
  def apply[T](l:List[T]): Queue[T] = ???
```
we could just do `Queue(0.25,0.1,0.15)`, that's what the usual libraries have